### ISSUE TITLE
ReDoS in debug@2.6.8 via Istanbul dependency

### ISSUE THREAD

**Issue by canterberry** [twuni/ante-boilerplate-js#5]
The following error occurs in CircleCI builds in the **security** job:

```
$ scripts/security
✗ Low severity vulnerability found on debug@2.6.8
- desc: Regular Expression Denial of Service (ReDoS)
- info: https://snyk.io/vuln/npm:debug:20170905
- from: ante-boilerplate-js@1.0.0 > istanbul-lib-source-maps@1.2.1 > debug@2.6.8
Your dependencies are out of date, otherwise you would be using a newer debug than debug@2.6.8.
Try deleting node_modules, reinstalling and running `snyk test` again.
If the problem persists, one of your dependencies may be bundling outdated modules.

Tested 140 dependencies for known vulnerabilities, found 1 vulnerability, 1 vulnerable path.

Run `snyk wizard` to address these issues.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exited with code 1
```

This fix has already been merged [here](https://github.com/istanbuljs/istanbuljs/commit/6f8a70ed833de1deb3574713f934b659345c8e1a), on 2017-09-28, and the `nyc` package has a tag for `v11.3.0` as of yesterday, but this version has not yet been published to npm and cannot be used directly as a `git+https` dependency due to the following warnings when attempting to install:

```
warning "nyc@11.3.0" is missing a bundled dependency "archy". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "arrify". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "caching-transform". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "convert-source-map". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "debug-log". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "default-require-extensions". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "find-cache-dir". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "find-up". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "foreground-child". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "glob". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "istanbul-lib-coverage". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "istanbul-lib-hook". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "istanbul-lib-report". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "istanbul-lib-source-maps". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "istanbul-reports". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "md5-hex". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "merge-source-map". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "micromatch". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "mkdirp". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "resolve-from". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "rimraf". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "signal-exit". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "spawn-wrap". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "test-exclude". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "yargs". This should be reported to the package maintainer.
warning "nyc@11.3.0" is missing a bundled dependency "yargs-parser". This should be reported to the package maintainer.
```

The only solution, it seems, would be to wait.

**Comment by canterberry** [comment]
Upgrading the debug package as a direct dependency of this project had no effect on the transitive vulnerable dependency.

**Comment by canterberry** [comment]
This issue was unintentionally closed when [PR #8](https://github.com/twuni/ante-boilerplate-js/pull/8) was merged; the intention was only to *reference* this issue, not to close it.
